src/analysis/calculus/inverse.lean

```diff
@@ -363,6 +379,32 @@ protected lemma inj_on (hf : approximates_linear_on f (f' : E →L[𝕜] F) s c)
   inj_on f s :=
 inj_on_iff_injective.2 $ hf.injective hc
 
+protected lemma surjective [complete_space E]
+  (hf : approximates_linear_on f (f' : E →L[𝕜] F) univ c) (hc : subsingleton E ∨ c < N⁻¹) :
+  surjective f :=
```

Comment:
Is it possible to give a more abstract proof along the following lines: by `antilipschitz_with.is_closed_range` the range of `f` is closed, and by `approximates_linear_on.open_image` the range of `f` is open?

Reply:
It's true this proof would work, but I feel it would be less honest: the fact that the image is open follows from the fact that the image of a ball of radius `R` contains a ball of radius `c R`, for some `c > 0`. And this fact obviously implies the surjectivity, modulo some quantifier gymnastic, so hiding this behind a connectedness argument would in my opinion hide what is really going on.

Reply:
OK, no problem.